src/main/webapp/app/tutor-exercise-dashboard/tutor-exercise-dashboard.component.ts

```diff
@@ -352,6 +364,10 @@ export class TutorExerciseDashboardComponent implements OnInit {
         this.router.navigate([route]);
     }
 
+    asProgrammingExercise(exercise: Exercise) {
```

Comment:
As a suggestion you can do it as a pipe to provide better performance and avoid calling this method each time https://stackoverflow.com/questions/45964576/type-casting-within-a-template-in-angular-2

Reply:
From performance it would be better, but we would have to create a type pipe for each exercise type as far as I'm aware. As this is the only case on the page it should be fine.